tree/tree/src/TChain.cxx

```diff
@@ -2524,6 +2524,12 @@ void TChain::ResetBranchAddresses()
    if (fTree) {
       fTree->ResetBranchAddresses();
    }
+   for (auto *frEl : TRangeDynCast<TFriendElement>(fFriends)) {
```

Comment:
As is, this will infinitely recurse in case there is a circular dependency in friendship (trivial reproducer `chain->AddFriend(friendchain); friendchain->AddFriend(chain);`)

And example of what needs to be done can be found in `TTree::SetBranchStatus` and the usage of `fFriendLockStatus` and `TFriendLock`. We will also need to add a new enum value in `TTree::ELockStatusBits`.

Also `TTree::ResetBranchAddresses()` needs to be similarly updated.  (The same example that fails with `TChain` will fail if the chains are replaced by trees)

Reply:
Thanks for the help and hints. I've tried now to follow that example.